ydb/core/tablet_flat/flat_executor.cpp

```diff
@@ -876,6 +876,10 @@ void TExecutor::ApplyFollowerUpdate(THolder<TEvTablet::TFUpdateBody> update) {
     if (update->IsSnapshot) // do nothing over snapshot after initial one
         return;
 
+    // Protect against recursive transactions in callbacks
+    Y_DEBUG_ABORT_UNLESS(!ActiveTransaction);
+    ActiveTransaction = true;
```

Comment:
It's hard to track that there is no `return` in that method before `ActiveTransaction = false` line

Maybe we can use some shared pointer, copy it here to a local variable and check that it has zero ref count, so it will be automatically released on field destruction? like lock_guard semantic

Reply:
Refactored into a `TActiveTransactionZone` RAII.